src/io/outputs/include/antares/io/outputs/columns.h

```diff
@@ -0,0 +1,188 @@
+    virtual std::string toString(size_t index) const = 0;
+    virtual size_t size() const = 0;
+    virtual void reserve(size_t capacity) = 0;
+    virtual void clear() = 0;
+};
+
+template<typename T>
+class TypedColumn: public IColumn
+{
+public:
+    explicit TypedColumn(std::function<std::string(const T&)> toStringFn = defaultToString,
+                         std::function<T(const std::string&)> fromStringFn = defaultFromString):
+        toStringFn_(toStringFn),
+        fromStringFn_(fromStringFn)
+    {
+    }
+
+    void add(const T& value)
+    {
+        data_.push_back(value);
+    }
+
+    void addFromString(const std::string& value) override
+    {
+        data_.push_back(fromStringFn_(value));
+    }
+
+    std::string toString(size_t index) const override
+    {
+        return toStringFn_(data_.at(index));
+    }
+
+    const T& get(size_t index) const
+    {
+        return data_.at(index);
+    }
+
+    size_t size() const override
+    {
+        return data_.size();
+    }
+
+    void reserve(size_t capacity) override
+    {
+        data_.reserve(capacity);
+    }
+
+    const std::vector<T>& data() const
+    {
+        return data_;
+    }
+
+    static std::string defaultToString(const T& v)
+    {
+        if constexpr (std::is_same_v<T, std::string>)
+        {
+            return v;
+        }
+        else
+        {
+            return std::to_string(v);
+        }
+    }
+
+    static T defaultFromString(const std::string& s)
+    {
+        if constexpr (std::is_same_v<T, std::string>)
+        {
+            return s;
+        }
+        else
+        {
+            T v;
+            std::istringstream iss(s);
+            iss >> v;
+            return v;
+        }
+    }
+
+    void clear() override
+    {
+        data_.clear();
+    }
+
+private:
+    std::vector<T> data_;
+    std::function<std::string(const T&)> toStringFn_;
+    std::function<T(const std::string&)> fromStringFn_;
+};
+
+using StringColumn = TypedColumn<std::string>;
+using IntColumn = TypedColumn<int>;
+
+class DoubleColumn: public TypedColumn<double>
+{
+public:
+    static std::string FromDouble(const double v)
+    {
+        std::ostringstream oss;
+        oss << std::setprecision(15) << v;
+        return oss.str();
+    }
+
+    static double ToDouble(const std::string& s)
+    {
+        return std::stod(s);
+    }
+
+    DoubleColumn():
+        TypedColumn(FromDouble, ToDouble)
+    {
+    }
+};
+
+template<typename T>
+class OptionalColumn: public TypedColumn<std::optional<T>>
+{
+public:
+    static std::string FromOptional(const std::optional<T>& option)
+    {
+        if (option.has_value())
+        {
+            if constexpr (std::is_floating_point_v<T>)
+            {
+                return DoubleColumn::FromDouble(option.value());
+            }
+            else
+            {
+                return TypedColumn<T>::defaultToString(option.value());
+            }
```

Comment:
Should be simpler with the `virtual` functions suggestion above

```suggestion
             return TypedColumn<T>::toString(option.value());
```

Reply:
that will turn TypedColumn::data_ into protected member and allow derived classes to modify it